lib/membrane_http_adaptive_stream/stream_sink_bin.ex

```diff
@@ -0,0 +1,146 @@
+                type: :time,
+                default: 2 |> Time.seconds()
+              ],
+              manifest_name: [
+                type: :string,
+                spec: String.t(),
+                default: "index",
+                description: "Name of the main manifest file"
+              ],
+              manifest_module: [
+                type: :atom,
+                spec: module,
+                description: """
+                Implementation of the `Membrane.HTTPAdaptiveStream.Manifest`
+                behaviour.
+                """
+              ],
+              storage: [
+                type: :struct,
+                spec: Storage.config_t(),
+                description: """
+                Storage configuration. May be one of `Membrane.HTTPAdaptiveStream.Storages.*`.
+                See `Membrane.HTTPAdaptiveStream.Storage` behaviour.
+                """
+              ],
+              target_window_duration: [
+                spec: pos_integer | :infinity,
+                type: :time,
+                default: Time.seconds(40),
+                description: """
+                Manifest duration is keept above that time, while the oldest segments
+                are removed whenever possible.
+                """
+              ],
+              persist?: [
+                type: :bool,
+                default: false,
+                description: """
+                If true, stale segments are removed from the manifest only. Once
+                playback finishes, they are put back into the manifest.
+                """
+              ],
+              target_segment_duration: [
+                type: :time,
+                default: 0,
+                description: """
+                Expected length of each segment. Setting it is not necessary, but
+                may help players achieve better UX.
+                """
+              ]
+
+  def_input_pad :input,
+    demand_unit: :buffers,
+    caps: [Membrane.H264, Membrane.AAC],
+    availability: :on_request,
+    options: [
+      encoding: [
+        spec: :H264 | :AAC,
+        description: """
+        Encoding atom determining which payloader will be used.
+        """
+      ]
+    ]
+
+  @impl true
+  def handle_init(opts) do
+    children = [
+      sink: %Sink{
+        manifest_name: opts.manifest_name,
+        manifest_module: opts.manifest_module,
+        storage: opts.storage,
+        target_window_duration: opts.target_window_duration,
+        persist?: opts.persist?,
+        target_segment_duration: opts.target_segment_duration
+      }
+    ]
+
+    {{:ok, spec: %ParentSpec{children: children, links: []}},
+     %{muxer_segment_duration: opts.muxer_segment_duration}}
+  end
+
+  @impl true
+  def handle_pad_added(Pad.ref(:input, ref) = pad, context, state) do
+    muxer_module = %MP4.CMAF.Muxer{segment_duration: state.muxer_segment_duration}
+
+    payloader_module =
+      case context.options[:encoding] do
```

Comment:
```suggestion
      case context.options.encoding do
```

Reply:
`options` is a list so `encoding` can't be accessed via dot